searxng/searxng
# Convert the infinite_scroll plugin as a preference (second version)

## What does this PR do?

Difference with #812 :

Use IntersectionObserver
* most probably fix https://github.com/searx/searx/issues/759
* for reference 
  * https://caniuse.com/?search=IntersectionObserver 
  * https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
  * a YT video about this subject: https://www.youtube.com/watch?v=2IbRtjez6ag

When there are only images, the `#sidebar` is hidden

## Why is this change important?

The infinite scroll is a long missing feature of the simple theme.

## How to test this PR locally?

Same protocol than PR #812 : 
* with different browser, at least Firefox and Chrom* : 
   * in the simple theme, and in the oscar theme:
     * check a request in the general category
     * check a request in the images category
     * check a request `!general !images time`

## Author's checklist

Extension: if `searxng.infinite_scroll_supported == false`, the preferences can disabled the infinite scroll preferences.

## Related issues

* close #812
* Part of https://github.com/searxng/searxng/issues/769
